### DISCUSSION TITLE
defer should be invoked when panicked

### DISCUSSION THREAD

**Discussion by i582** [vlang/v#18637] [Ideas]
### Describe the feature

```v
fn foo() int {
	defer {
		println(100)
	}
	
	panic("oops")
}

println(foo())
// 100
// panic ...
```

### Use Case

Close file descriptors gracefully in case of panic.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

 V 0.3.2 6d223b9.5aad0db

### Environment details (OS name and version, etc.)

```
OS: macos, macOS, 13.0.1, 22A400
Processor: 10 cpus, 64bit, little endian, Apple M1 Pro
CC version: Apple clang version 14.0.0 (clang-1400.0.29.202)

getwd: /Users/petrmakhnev/vlang-foundation/specification
vmodules: /Users/petrmakhnev/.vmodules
vroot: /Users/petrmakhnev/v
vexe: /Users/petrmakhnev/v/v
vexe mtime: 2023-01-24 17:38:21
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 6d223b9.5aad0db

Git version: git version 2.37.1 (Apple Git-137.1)
Git vroot status: weekly.2023.04-1-gfc655775-dirty (2 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: pmakhnev/fix_operator_overloading_doc 173c526e
```

**Comment by dumblob** [comment]
This is actually more complicated than it seems:

https://github.com/vlang/v/issues/9388
https://github.com/vlang/v/issues/2030

**Comment by i582** [comment]
@dumblob thank you for links!

**Comment by JalonSolov** [comment]
This is why I keep saying `panic()` should never be used unless it's to avoid something completely catastrophic, like wiping the user's drives, etc.

It is an immediate exit, no cleanup, get out NOW operation.
